cjdns-core/src/encoding.rs

```diff
@@ -234,28 +232,30 @@ fn read_n_bits_from_position_into_u32(vecc: &Vec<u8>, position: u32, bits_amount
 ///
 /// Accepts bytes array, parses it and returns vector of `EncodingSchemeForm`s.
 pub fn deserialize_forms(vecbytes: &Vec<u8>) -> Result<Vec<EncodingSchemeForm>, Error> {
+    let mut reversed_input_bytes = vecbytes.clone();
+    reversed_input_bytes.reverse();
```

Comment:
Not good. Cloning input array requires an allocation. This is definitely not necessary in this case.

I'd suggest to use reversed iterator here: `vecc.iter().rev()` which is more that sufficient. Also argument name could be better (maybe `data` instead of `vecc`).